pkg/kubectl/cmd/patch.go

```diff
@@ -191,17 +192,8 @@ func RunPatch(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 			}
 			count++
 
-			oldData, err := json.Marshal(info.Object)
-			if err != nil {
-				return err
-			}
-			newData, err := json.Marshal(patchedObj)
-			if err != nil {
-				return err
-			}
-			if !reflect.DeepEqual(oldData, newData) {
+			if !reflect.DeepEqual(info.Object, patchedObj) {
```

Comment:
```go
didPatch := !reflect.DeepEqual(info.Object, patchedObj)
```
And make the `didPatch` variable local to each scope.

Reply:
Done.